<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>288</width>
    <height>485</height>
   </rect>
  </property>
  <property name="maximumSize">
   <size>
    <width>288</width>
    <height>16777215</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Arduino Console</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <property name="spacing">
     <number>10</number>
    </property>
    <property name="leftMargin">
     <number>10</number>
    </property>
    <property name="topMargin">
     <number>10</number>
    </property>
    <property name="rightMargin">
     <number>10</number>
    </property>
    <property name="bottomMargin">
     <number>10</number>
    </property>
    <item>
     <layout class="QVBoxLayout" name="verticalLayout">
      <property name="spacing">
       <number>10</number>
      </property>
      <item>
       <widget class="QTextBrowser" name="message_browser">
        <property name="minimumSize">
         <size>
          <width>266</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>266</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="html">
         <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'SimSun'; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;For devices described as Arduino.&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; font-weight:600;&quot;&gt;&lt;br /&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;Setting Example&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-weight:600; color:#ffaa00;&quot;&gt;Serial&lt;/span&gt;&lt;span style=&quot; color:#ffaa00;&quot;&gt;.begin&lt;/span&gt;(19200,SERIAL_8E2);&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;Baud Rate: 19200&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;Data Bits: 8&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;Stop Bits: 2&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;Parity: Even Parity&lt;/p&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;br /&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;Arduino Console Mannual:&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;Type and press enter to send message.&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; color:#000000;&quot;&gt;Click Load to load work flow.&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; color:#000000;&quot;&gt;On: Start workflow&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; color:#000000;&quot;&gt;Off; Stop workflow&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; color:#000000;&quot;&gt;||: Pause&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&amp;gt;|: Execute next step without waiting.&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&amp;lt; : Step back.&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&amp;gt; : Step forward.&lt;/p&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;br /&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;JSON workflow Example&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;{&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;  &lt;span style=&quot; color:#00aaff;&quot;&gt;&amp;quot;Description&amp;quot;&lt;/span&gt;:&lt;span style=&quot; color:#ffaa00;&quot;&gt;&amp;quot;Example&amp;quot;&lt;/span&gt;,&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;  &lt;span style=&quot; color:#55aaff;&quot;&gt;&amp;quot;Workflow&amp;quot;&lt;/span&gt;:[&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;    {&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;      &lt;span style=&quot; color:#00aaff;&quot;&gt;&amp;quot;Operation&amp;quot;&lt;/span&gt;:&lt;span style=&quot; color:#ffaa00;&quot;&gt;&amp;quot;xxxx&amp;quot;&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;      &lt;span style=&quot; color:#00aaff;&quot;&gt;&amp;quot;Time&amp;quot;&lt;/span&gt;:&lt;span style=&quot; color:#00aa00;&quot;&gt;0.5&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;      &lt;span style=&quot; color:#55aaff;&quot;&gt;&amp;quot;Commands&amp;quot;&lt;/span&gt;:[&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;        &lt;span style=&quot; color:#ffaa00;&quot;&gt;&amp;quot;xxxx&amp;quot;&lt;/span&gt;&lt;span style=&quot; color:#000000;&quot;&gt;,&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; color:#ffaa00;&quot;&gt;        &amp;quot;xxxx&amp;quot;&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;      ]&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;    },&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;    {&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;      &lt;span style=&quot; color:#00aaff;&quot;&gt;&amp;quot;Operation&amp;quot;&lt;/span&gt;:&lt;span style=&quot; color:#ffaa00;&quot;&gt;&amp;quot;xxxx&amp;quot;&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;      &lt;span style=&quot; color:#00aaff;&quot;&gt;&amp;quot;Time&amp;quot;&lt;/span&gt;:&lt;span style=&quot; color:#00aa00;&quot;&gt;100&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;      &lt;span style=&quot; color:#55aaff;&quot;&gt;&amp;quot;Commands&amp;quot;&lt;/span&gt;:[&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;        &lt;span style=&quot; color:#ffaa00;&quot;&gt;&amp;quot;xxxx&amp;quot;&lt;/span&gt;&lt;span style=&quot; color:#000000;&quot;&gt;,&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; color:#ffaa00;&quot;&gt;        &amp;quot;xxxx&amp;quot;&lt;/span&gt;&lt;span style=&quot; color:#000000;&quot;&gt;,&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; color:#ffaa00;&quot;&gt;        &amp;quot;xxxx&amp;quot;&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;      ]&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;    }&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;  ]&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;}&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; color:#55aaff;&quot;&gt;&amp;quot;Workflow&amp;quot;&lt;/span&gt;&lt;span style=&quot; color:#000000;&quot;&gt; and &lt;/span&gt;&lt;span style=&quot; color:#55aaff;&quot;&gt;&amp;quot;Commands&amp;quot;&lt;/span&gt;&lt;span style=&quot; color:#000000;&quot;&gt; are scalable&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; color:#000000;&quot;&gt;For each workflow, it will wait &lt;/span&gt;&lt;span style=&quot; color:#00aaff;&quot;&gt;&amp;quot;Time&amp;quot;&lt;/span&gt;&lt;span style=&quot; color:#000000;&quot;&gt; seconds to run next, while the commands are sent in batch.&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; color:#000000;&quot;&gt;&lt;br /&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QTabWidget" name="arduino_control_widget">
        <property name="minimumSize">
         <size>
          <width>266</width>
          <height>191</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>266</width>
          <height>191</height>
         </size>
        </property>
        <property name="currentIndex">
         <number>0</number>
        </property>
        <widget class="QWidget" name="comport_console_panel">
         <attribute name="title">
          <string>Console</string>
         </attribute>
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <property name="spacing">
           <number>10</number>
          </property>
          <property name="leftMargin">
           <number>10</number>
          </property>
          <property name="topMargin">
           <number>10</number>
          </property>
          <property name="rightMargin">
           <number>10</number>
          </property>
          <property name="bottomMargin">
           <number>10</number>
          </property>
          <item>
           <widget class="QLabel" name="manual_cli_label">
            <property name="text">
             <string>----- Manual Command Line -----</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="command_line_edit"/>
          </item>
          <item>
           <layout class="QHBoxLayout" name="manual_panel">
            <property name="spacing">
             <number>6</number>
            </property>
            <item>
             <widget class="QPushButton" name="port_switch_button">
              <property name="text">
               <string>Open Port</string>
              </property>
              <property name="checkable">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="clear_message_button">
              <property name="text">
               <string>Clear Messages</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <spacer name="verticalSpacer">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QLabel" name="auto_executor_label">
            <property name="text">
             <string>-- Automatic Command Executor --</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="auto_executor_panel">
            <property name="spacing">
             <number>6</number>
            </property>
            <item>
             <widget class="QPushButton" name="load_button">
              <property name="minimumSize">
               <size>
                <width>55</width>
                <height>0</height>
               </size>
              </property>
              <property name="text">
               <string>Load</string>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="automation_buttons">
              <property name="spacing">
               <number>0</number>
              </property>
              <property name="sizeConstraint">
               <enum>QLayout::SetNoConstraint</enum>
              </property>
              <item>
               <widget class="QPushButton" name="start_button">
                <property name="enabled">
                 <bool>true</bool>
                </property>
                <property name="minimumSize">
                 <size>
                  <width>0</width>
                  <height>0</height>
                 </size>
                </property>
                <property name="maximumSize">
                 <size>
                  <width>30</width>
                  <height>16777215</height>
                 </size>
                </property>
                <property name="text">
                 <string>On</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="stop_button">
                <property name="enabled">
                 <bool>false</bool>
                </property>
                <property name="minimumSize">
                 <size>
                  <width>0</width>
                  <height>0</height>
                 </size>
                </property>
                <property name="maximumSize">
                 <size>
                  <width>30</width>
                  <height>16777215</height>
                 </size>
                </property>
                <property name="text">
                 <string>Off</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="move_iterator">
              <property name="spacing">
               <number>0</number>
              </property>
              <item>
               <widget class="QPushButton" name="pause_button">
                <property name="enabled">
                 <bool>false</bool>
                </property>
                <property name="maximumSize">
                 <size>
                  <width>30</width>
                  <height>16777215</height>
                 </size>
                </property>
                <property name="text">
                 <string>||</string>
                </property>
                <property name="checkable">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="skip_work_button">
                <property name="enabled">
                 <bool>false</bool>
                </property>
                <property name="text">
                 <string>&gt;|</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="last_work_button">
                <property name="enabled">
                 <bool>false</bool>
                </property>
                <property name="maximumSize">
                 <size>
                  <width>30</width>
                  <height>16777215</height>
                 </size>
                </property>
                <property name="text">
                 <string>&lt;</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="next_work_button">
                <property name="enabled">
                 <bool>false</bool>
                </property>
                <property name="maximumSize">
                 <size>
                  <width>30</width>
                  <height>16777215</height>
                 </size>
                </property>
                <property name="text">
                 <string>&gt;</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="comport_setting_panel">
         <attribute name="title">
          <string>Setting</string>
         </attribute>
         <layout class="QVBoxLayout" name="verticalLayout_4">
          <property name="leftMargin">
           <number>10</number>
          </property>
          <property name="topMargin">
           <number>10</number>
          </property>
          <property name="rightMargin">
           <number>10</number>
          </property>
          <property name="bottomMargin">
           <number>10</number>
          </property>
          <item>
           <layout class="QGridLayout" name="gridLayout">
            <property name="spacing">
             <number>10</number>
            </property>
            <item row="3" column="1">
             <widget class="QComboBox" name="databits_box">
              <property name="currentText">
               <string>5</string>
              </property>
              <item>
               <property name="text">
                <string>5</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>6</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>7</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>8</string>
               </property>
              </item>
             </widget>
            </item>
            <item row="4" column="0">
             <widget class="QLabel" name="stopbits_label">
              <property name="text">
               <string>Stop Bits</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignCenter</set>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="baudrate_label">
              <property name="text">
               <string>Baud Rate</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignCenter</set>
              </property>
             </widget>
            </item>
            <item row="5" column="0">
             <widget class="QLabel" name="parity_label">
              <property name="text">
               <string>Parity</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignCenter</set>
              </property>
             </widget>
            </item>
            <item row="4" column="1">
             <widget class="QComboBox" name="stopbits_box">
              <property name="currentIndex">
               <number>1</number>
              </property>
              <item>
               <property name="text">
                <string>1 - One Stop</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>2 - Two Stop</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>3 - One And Half Stop</string>
               </property>
              </item>
             </widget>
            </item>
            <item row="5" column="1">
             <widget class="QComboBox" name="parity_box">
              <property name="currentIndex">
               <number>1</number>
              </property>
              <item>
               <property name="text">
                <string>0 - No Parity</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>2 - Even Parity</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>3 - Odd Parity</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>4 - Space Parity</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>5 - Mark Parity</string>
               </property>
              </item>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QComboBox" name="portname_box"/>
            </item>
            <item row="2" column="1">
             <widget class="QComboBox" name="baudrate_box">
              <property name="currentText">
               <string>19200</string>
              </property>
              <property name="currentIndex">
               <number>4</number>
              </property>
              <item>
               <property name="text">
                <string>1200</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>2400</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>4800</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>9600</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>19200</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>38400</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>57600</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>115200</string>
               </property>
              </item>
             </widget>
            </item>
            <item row="3" column="0">
             <widget class="QLabel" name="databits_label">
              <property name="text">
               <string>Data Bits</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignCenter</set>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QPushButton" name="refresh_available_arduino">
              <property name="minimumSize">
               <size>
                <width>0</width>
                <height>0</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>55</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="text">
               <string>Refresh</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
